### PR TITLE
docs: sync install-dependencies.md & polish it

### PR DESCRIPTION
**Pull request by spacewander**
Signed-off-by: spacewander <spacewanderlzx@gmail.com>

### Pre-submission checklist:

* [x] Did you explain what problem does this PR solve? Or what new features have been added?
* [ ] Have you added corresponding test cases?
* [x] Have you modified the corresponding document?
* [ ] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix/tree/master#community) first**

### CODE REVIEW

**Review by Yiyiyimu — APPROVED** [review]

File: doc/zh-cn/install-dependencies.md [hunk 1]
```diff
@@ -54,7 +52,10 @@ sudo yum install yum-utils
 sudo yum-config-manager --add-repo https://openresty.org/package/centos/openresty.repo
 
 # 安装 OpenResty 和 编译工具
-sudo yum install -y openresty curl git gcc luarocks lua-devel
+sudo yum install -y openresty curl git gcc lua-devel openresty-openssl-devel
```
Comment (membphis) [inline doc/zh-cn/install-dependencies.md:55]: can we use the `luajit` of openresty? if yes, we do not need to install the `lua-devel`.

**Review by membphis — APPROVED** [review]

**Review by membphis — CHANGES_REQUESTED** [review]
cancel my approve

File: doc/zh-cn/install-dependencies.md (hunk 1 above)
Comment (spacewander) [reply doc/zh-cn/install-dependencies.md]: Updated.

**Comment by Yiyiyimu** [comment]
When I follow the installation steps and install luarocks with script, I got the following error:
```
Configuring LuaRocks version 3.4.0...

Lua interpreter not found in $PATH
You may want to use the flags --with-lua, --with-lua-bin and/or --lua-version
Run ./configure --help for details.

configure failed.

Please run the ./configure script before building.

make: *** No rule to make target 'config.unix', needed by 'luarocks'.  Stop.
Please run the ./configure script before building.

make: *** No rule to make target 'config.unix', needed by 'build/luarocks'.  Stop.
mkdir: cannot create directory ‘/home/shuyang/.luarocks’: File exists
Wrote
        variables.OPENSSL_LIBDIR = "/usr/local/openresty/openssl/lib"
to
        /usr/local/etc/luarocks/config-5.1.lua
Wrote
        variables.OPENSSL_INCDIR = "/usr/local/openresty/openssl/include"
to
        /usr/local/etc/luarocks/config-5.1.lua
```

Environment: Ubuntu 18.04, on Windows (WSL2)

**Comment by spacewander** [comment]
The `linux-install-openresty.sh` is wrong. It misses `--with-lua=$OPENRESTY_PREFIX/luajit`

File: doc/install-dependencies.md [hunk 2]
```diff
@@ -35,15 +35,13 @@
...
+- If it is OpenResty 1.19, APISIX will use OpenResty's built-in LuaJIT to run `bin/apisix`; otherwise it will use Lua 5.1. If you encounter `luajit: lj_asm_x86.h:2819: asm_loop_ fixup: Assertion '((intptr_t)target & 15) == 0' failed`, this is a problem with the low version of OpenResty's built-in LuaJIT under certain compilation conditions.
+
+- - On some platforms, installing LuaRocks via the package manager will cause Lua to be upgraded to Lua 5.3, so we recommend installing LuaRocks via source code. if you install OpenResty and its OpenSSL develop library (openresty-openssl-devel for rpm and openresty-openssl-dev for deb) via the official repository, then [we provide a script for automatic installation](. /... /utils/linux-install-luarocks.sh). If you compile OpenResty yourself, you can refer to the above script and change the path in it. If you don't specify the OpenSSL library path when you compile, you don't need to configure the OpenSSL variables in LuaRocks, because the system's OpenSSL is used by default. If the OpenSSL library is specified at compile time, then you need to ensure that LuaRocks' OpenSSL configuration is consistent with OpenResty's.
```
Comment (tokers) [inline doc/install-dependencies.md:40]: We may also give some tips about how to configuring the OpenSSL configuration for LuaRocks.

Comment (spacewander) [reply doc/install-dependencies.md:40]: Read the script is enough. Should not repeat yourself.

Comment (tokers) [reply doc/install-dependencies.md:40]: OK.

**Review by membphis — APPROVED** [review]

**Pull request merged**